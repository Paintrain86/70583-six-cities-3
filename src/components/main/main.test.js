import React from 'react';
import renderer from 'react-test-renderer';
import Main from './main.jsx';

describe(`Main`, () => {
  it(`Should be rendered correctly`, () => {
    const testSettings = {
      offers: [
        {
          id: 1,
          title: `Little cozy trash can`,
          image: `https://cdnmedia.220-volt.ru/content/products/337/337834/images/thumb_220/n1200x800_q80/1.jpeg`
        },
        {
          id: 2,
          title: `Amazing hotel beach resort`,
          image: `https://img1.coastalliving.timeinc.net/sites/default/files/styles/4_3_horizontal_inbody_900x506/public/image/2016/11/main/coastalhotels_0215_09.jpg`
        },
        {
          id: 3,
          title: `Famous japanese boxes`,
          image: ``
        }
      ],
      onOfferClick: jest.fn()
    };

    const markup = renderer
      .create(<Main {...testSettings}/>)
      .toJSON();

    expect(markup).toMatchSnapshot();
  });
});
