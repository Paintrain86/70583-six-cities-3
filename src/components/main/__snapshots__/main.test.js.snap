// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Should be rendered correctly 1`] = `
<div
  className="offers"
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <div
    className="offers__item"
    style={
      Object {
        "width": "30%",
      }
    }
  >
    <img
      alt="Little cozy trash can"
      className="offers__item-img"
      src="https://cdnmedia.220-volt.ru/content/products/337/337834/images/thumb_220/n1200x800_q80/1.jpeg"
      style={
        Object {
          "maxWidth": "100%",
        }
      }
    />
    <h4
      className="offers__item-title"
      onClick={[MockFunction]}
    >
      Little cozy trash can
    </h4>
  </div>
  <div
    className="offers__item"
    style={
      Object {
        "width": "30%",
      }
    }
  >
    <img
      alt="Amazing hotel beach resort"
      className="offers__item-img"
      src="https://img1.coastalliving.timeinc.net/sites/default/files/styles/4_3_horizontal_inbody_900x506/public/image/2016/11/main/coastalhotels_0215_09.jpg"
      style={
        Object {
          "maxWidth": "100%",
        }
      }
    />
    <h4
      className="offers__item-title"
      onClick={[MockFunction]}
    >
      Amazing hotel beach resort
    </h4>
  </div>
  <div
    className="offers__item"
    style={
      Object {
        "width": "30%",
      }
    }
  >
    <img
      alt="Famous japanese boxes"
      className="offers__item-img"
      src="https://dvnak.ru/upload/old/photos/medium/bbe7280e8e0d144e5e7710286e7ccc45.jpg"
      style={
        Object {
          "maxWidth": "100%",
        }
      }
    />
    <h4
      className="offers__item-title"
      onClick={[MockFunction]}
    >
      Famous japanese boxes
    </h4>
  </div>
</div>
`;
